crates/pixi_manifest/src/script_metadata.rs

```diff
@@ -0,0 +1,419 @@
+            ),
+            "osx" => matches!(platform, Platform::Osx64 | Platform::OsxArm64),
+            "win" => matches!(platform, Platform::Win64 | Platform::WinArm64),
+            specific => {
+                // Try to parse as a specific platform
+                if let Ok(specific_platform) = Platform::from_str(specific) {
+                    specific_platform == platform
+                } else {
+                    false
+                }
+            }
+        }
+    }
+}
+
+/// Extract the conda-script metadata block from a reader
+fn extract_metadata_block<R: BufRead>(reader: R) -> Result<String, ScriptMetadataError> {
+    let mut in_block = false;
+    let mut toml_lines = Vec::new();
+    let mut comment_prefix: Option<String> = None;
+
+    for line in reader.lines() {
+        let line = line?;
+        let trimmed = line.trim_start();
+
+        // Detect the start of the block
+        if !in_block && trimmed.contains("/// conda-script") {
+            in_block = true;
+            // Detect comment syntax (e.g., "#", "//", "--")
+            comment_prefix = detect_comment_prefix(&line);
+            continue;
+        }
+
+        // Detect the end of the block
+        if in_block && trimmed.contains("/// end-conda-script") {
+            return Ok(toml_lines.join("\n"));
+        }
+
+        // Extract TOML content inside the block
+        if in_block {
+            if let Some(ref prefix) = comment_prefix {
+                if let Some(content) = line.strip_prefix(prefix) {
+                    // Remove the comment prefix and add to TOML content
+                    toml_lines.push(content.trim_start().to_string());
+                } else if trimmed.is_empty() {
+                    // Allow empty lines
+                    toml_lines.push(String::new());
+                } else {
+                    return Err(ScriptMetadataError::MalformedBlock(format!(
+                        "Expected line to start with '{}', got: {}",
+                        prefix, line
+                    )));
+                }
+            }
+        }
+    }
+
+    if in_block {
+        return Err(ScriptMetadataError::MalformedBlock(
+            "Metadata block started but never closed with '/// end-conda-script'".to_string(),
+        ));
+    }
+
+    Err(ScriptMetadataError::NoMetadataFound)
+}
+
+/// Detect the comment prefix used in the script
+fn detect_comment_prefix(line: &str) -> Option<String> {
+    // Common comment prefixes
+    let prefixes = ["# ", "// ", "-- ", "/* "];
+
+    for prefix in &prefixes {
+        if line.trim_start().starts_with(prefix) {
+            return Some(prefix.to_string());
+        }
+    }
+
+    // Also support without trailing space
+    let prefixes_no_space = ["#", "//", "--"];
+    for prefix in &prefixes_no_space {
+        if line.trim_start().starts_with(prefix) {
+            return Some(prefix.to_string());
+        }
+    }
+
+    None
```

Comment:
Whats the difference between trailing space or not? There seems to be duplication here.